lib/balance-tx/lib/internal/Internal/Cardano/Write/Tx/Balance.hs

```diff
@@ -876,32 +875,18 @@ balanceTransactionWithSelectionStrategyAndNoZeroAdaAdjustment
             minfee' = Convert.toLedgerCoin minfee
         return (balance, minfee', witCount)
 
-    -- | Ensure the wallet UTxO is consistent with a provided @CardanoApi.UTxO@.
+    -- | Ensure the wallet UTxO is consistent with the given UTxO.
     --
     -- They are not consistent iff an input can be looked up in both UTxO sets
     -- with different @Address@, or @TokenBundle@ values.
```

Comment:
I fear this is no longer respected. 

The point of it is to allowing calling the `transactions-balance` API endpoint with inputs in the wallet which have been extended with datums or inlines scripts, even though the wallet is unable tracking them correctly.

Or said differently, this ensures procedures like
```haskell
setupAndUseReferenceScriptAtAddress :: Address -> IO ()
setupAndUseReferenceScriptAtAddress addr = do
    1. Have wallet create an inline script at `addr`
    2. Reference the input and use the script, calling transactions-balance manually providing the reference utxo for lookup
```

works for all values of `addr` (as opposed to breaking if `addr` belongs to the wallet.)

Reply:
@Anviking Okay -- I've adjusted the code so that a conflict is only reported if the `Address` or `TokenBundle` parts are different.